subsys/net/l2/openthread/openthread.c

```diff
@@ -361,7 +431,13 @@ static void openthread_start(struct openthread_context *ot_context)
 		/* No dataset - initiate network join procedure. */
 		NET_DBG("Starting OpenThread join procedure.");
 
-		error = otJoinerStart(ot_instance, OT_JOINER_PSKD, NULL,
+		/* should this fail, we'll stick to OT_JOINER_PSKD */
+		openthread_get_pskd(&joiner_pskd);
+		u32_t join_delay = sys_rand32_get();
```

Comment:
What's with those delays? We can't hold up the entire system for up to 15 seconds. The commit message does not mention it at all.

Reply:
I accidentally pushed that part too but that should not be in here. There is a different open pull request for this so i will remove it here.